openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -578,19 +583,31 @@ def post(self, request, course_key_string, *args, **kwargs):
 
         data = dict(request.data)
         first_section = data.pop('first_section', None)
+        first_discussion = data.pop('first_discussion', None)
         if data:
             return Response(status=400)  # there were parameters we didn't recognize
 
-        enrollment = CourseEnrollment.get_enrollment(request.user, course_key)
-        if not enrollment:
-            return Response(status=404)
-
+        object_created = False
+        user_celebration = False
+        enrollment_celebration = False
         defaults = {}
```

Comment:
So yeah, this is weird, right? I'm using one `defaults` dictionary for two models. I did some local testing of what happens in this scenario (passing in a field that doesn't exist on your model), and it seems like it will end up fine. It will properly save the fields that match, and for the fields that don't match, it will add them as attributes to this instance (which is fine in this case because we then throw it away when the function returns).

Let me know your thoughts and I can do two default dictionaries. I was just trying to save some code (and potentially account for the case where we use the same field name on both enrollment and user.. I feel like that shouldn't happen because the use cases, but I dunno. It's a double-edged sword I suppose)

Reply:
I'd personally vote for separate defaults. Like, even if we use the same field name on both, it's not necessarily true that they should have the same value?

Honestly actualy, maybe we shouldn't set the user-celebrations in this same endpoint, which requires a course id. Like, some user celebrations might be done outside the context of a course and then what endpoint would an MFE hit?